src/Host/Client/Impl/Host/RHostConnector.cs

```diff
@@ -83,35 +107,62 @@ internal abstract class RHostConnector : IRHostConnector {
             await ConnectToBrokerAsync();
 
             rCommandLineArguments = rCommandLineArguments ?? string.Empty;
-
-            try {
-                var sessions = new SessionsWebService(_broker);
-                await sessions.PutAsync(name, new SessionCreateRequest {
-                    InterpreterId = _interpreterId,
-                    CommandLineArguments = rCommandLineArguments,
-                });
-            } catch (HttpRequestException ex) {
-                // If UICredentials was canceled by the user, we'll get HttpRequestException(WebException(OperationCanceledException)) here.
-                // Unwrap all the extraneous layers, and rethrow the cancellation.
-                var oce = ex?.InnerException?.InnerException as OperationCanceledException;
-                if (oce != null) {
-                    throw oce;
+            var sessions = new SessionsWebService(HttpClient);
+
+            while (true) {
+                bool isValidCredentials = true;
+                try {
+                    UpdateCredentials();
```

Comment:
Good catch. It's harmless with the current implementation, but it's not what I had in mind.